### DISCUSSION TITLE
check intent source

### DISCUSSION THREAD

**Discussion by GuilhermeDeLucca** [LiveHelperChat/livehelperchat#1812] [General]
remdex, I would like to ask you for guidance/opinion

I use RASA to see my Intents.

And basically everything always leads me to the next flow.

LHC makes a CRM REST API

1. Collect User ID

2. Returns the user ID contracts.
 
1 and 2 standard flow (always happens)

3. N flows can happen here, it will depend on the original intet for example to perform a new CRM REST API

Intent Pay bill
flow A

Intert Track order
Flow B

My question is:

from step 3 you could use (Check for conditions to proceed) to redirect to REST API, so how would it be correct to do that? Should I create a variable (IntentName) and always update it to perform (Check for conditions to proceed) or is there a smarter way to do this?

**Comment by remdex** [comment]
I'm not sure what exactly you are asking. Bu your flow should follow for intent parsing.
https://doc.livehelperchat.com/docs/bot/rasa-integration-intent

Also it can be that once you ask how can I help you further you have something like this 

![image](https://user-images.githubusercontent.com/1146085/202478904-b5890320-ff00-4ddc-921e-3959800f720c.png)

And intent parser looks like.

![image](https://user-images.githubusercontent.com/1146085/202479330-ede3844e-de0a-4ae4-aaaf-47b8f6e61a3f.png)

**Comment by remdex** [comment]
Also with intent parser triggers you just have to define events section. In this case intent is `greet` returned from rasa.
![image](https://user-images.githubusercontent.com/1146085/202481871-5f2b1a50-7d42-4c1e-b39f-d2b309bcf406.png)

**Comment by GuilhermeDeLucca** [comment]
Here I made the variable I told you (I don't know, it's a good practice)

![Captura de Tela 2022-11-17 às 12 46 45](https://user-images.githubusercontent.com/9898902/202492838-99cdeea5-775b-4382-aa2a-6e2987cdddbc.png)


Here I do the validation (the client will always have to identify himself)

So I can be very accurate in the intent request

![Captura de Tela 2022-11-17 às 12 46 24](https://user-images.githubusercontent.com/9898902/202494796-b75d7295-fd71-428c-9b11-ffc368742403.png)


NOTE: This chat is not working, but I know that in theory it should

**Reply by remdex** [reply]
Instead of that workflow you should have just Intent parser trigger as per example.
![image](https://user-images.githubusercontent.com/1146085/202645907-7537d0c0-2423-4754-a641-d7783504d366.png)
